docs/Guide/plugins/API/rate-limiting.mdx

```diff
@@ -3,4 +3,133 @@ title: RateLimiting routes
+
+    		return true;
+    	},
+    	(_request, response) => {
+    		response.error(HttpCodes.TooManyRequests);
+    	}
+        );
+
+}
+
+````
+</TabItem>
+
+<TabItem value="typescript" label="TypeScript">
+  ```ts
+  import { createFunctionPrecondition } from '@sapphire/decorators';
+import { ApiRequest, ApiResponse, HttpCodes } from '@sapphire/plugin-api';
+import { RateLimitManager } from '@sapphire/ratelimits';
+
+export function ratelimit(time: number, limit = 1, auth = false) {
+	const manager = new RateLimitManager(time, limit);
+	const xRateLimitLimit = time;
+	return createFunctionPrecondition(
+		(request: ApiRequest, response: ApiResponse) => {
+			const id = (auth ? request.auth!.id : request.headers['x-api-key'] || request.connection.remoteAddress) as string;
+			const bucket = manager.acquire(id);
+
+			response.setHeader('Date', new Date().toUTCString());
+			if (bucket.limited) {
+				response.setHeader('Retry-After', bucket.remainingTime.toString());
+				return false;
+			}
+
+			try {
+				bucket.consume();
+			} catch {}
+
+			response.setHeader('X-RateLimit-Limit', xRateLimitLimit);
+			response.setHeader('X-RateLimit-Remaining', bucket.remaining.toString());
+			response.setHeader('X-RateLimit-Reset', bucket.remainingTime.toString());
+
+			return true;
+		},
+		(_request: ApiRequest, response: ApiResponse) => {
+			response.error(HttpCodes.TooManyRequests);
+		}
+      );
+}
+  ```
+</TabItem>
+
+You then want to make a `seconds` function.
+<Tabs groupId="lang choice">
+  <TabItem value="javascript" label="JavaScript">
+    ```js
+    TODO
+    ```
+  </TabItem>
+
+  <TabItem value="typescript" label="TypeScript">
+    ```ts
+    import { Time } from '@sapphire/time-utilities';
+
+/**
+* Converts a number of seconds to milliseconds.
+* @param seconds The amount of seconds
+* @returns The amount of milliseconds `seconds` equals to.
+*/
+export function seconds(seconds: number): number {
+	return seconds * Time.Second;
+}
+    ```
+  </TabItem>
+
+ ## Now Actually Protecting The Routes
+  Protecting the routes is pretty easy, and can be done so by doing
+
+  <Tabs groupId="lang choice">
+    <TabItem value="javascript" label="JavaScript">
+      ```js
+      TODO
+      ```
+
+    </TabItem>
+
+    <TabItem value="typescript" label="TypeScript">
+      ```ts
+      import { ApplyOptions } from '@sapphire/decorators';
+      import { ApiRequest, ApiResponse, methods, Route, RouteOptions } from '@sapphire/plugin-api';
+      import { check } from '../scripts/check'
+      import { GifsModel } from '../Schemas/gifs';
+      import uuid4 from 'uuid4'
+      import { ratelimit } from '../lib/api/utils';
+      import { seconds } from '../lib/utils/common/seconds';
+
+@ApplyOptions<RouteOptions>({ route: '/api/hug' })
+export class UserRoute extends Route {
+  @ratelimit(seconds(1), 1, false)
+	public async [methods.GET](_request: ApiRequest, response: ApiResponse) {
+````
```

Comment:
- Indenting is way off here
- Closes with 5 back ticks
- JS cannot use decorators without babel preprocessing
- Missing TS
- Missing ESM
- The header is horrible here. It should actually be a header, not a following step.